dubbo-registry/dubbo-registry-nacos/src/main/java/org/apache/dubbo/registry/nacos/NacosNamingServiceWrapper.java

```diff
@@ -269,14 +244,9 @@ public void deregisterInstance(String serviceName, String group, Instance instan
             InstanceInfo instanceInfo = optional.get();
             instancesInfo.getInstances().remove(instanceInfo);
 
-            try {
-                mapLock.lock();
-                if (instancesInfo.getInstances().isEmpty()) {
-                    registerStatus.remove(new InstanceId(nacosServiceName, group));
-                    instancesInfo.setValid(false);
-                }
-            } finally {
-                mapLock.unlock();
+            if (instancesInfo.getInstances().isEmpty()) {
+                registerStatus.remove(new InstanceId(nacosServiceName, group));
+                instancesInfo.setValid(false);
```

Comment:
Operation here is not atomic

Reply:
@AlbumenJ 
I think there is no need for atomicity guarantee here.
- because the code here has already got instancesInfo lock ```instancesInfo.lock()```; 
- and for the modification of registerStatus ```registerStatus.remove(new InstanceId(nacosServiceName, group));```, due to the protection of the lock mechanism inside concurrenthashmap , which has the same effect as the previous mapLock.lock()